### ISSUE TITLE
Consider adding SMB to port blacklist

### ISSUE THREAD

**Issue by zackw** [whatwg/fetch#544]
Please consider adding the ports used for Microsoft's SMB protocol (139, 445) to the "bad port" blacklist.  This is prompted by the [MS17-010](https://github.com/RiskSense-Ops/MS17-010) exploit, but there's quite a long history of remotely exploitable SMB bugs.  This will, at least, make it harder to [scan for exploitable systems](http://www.andlabs.org/tools/jsrecon/jsrecon.html) from a malicious website.

**Comment by annevk** [comment]
cc @mikewest

**Comment by mikewest** [comment]
This seems like a reasonable thing to explore. I can add some metrics to Chrome. @zackw: Would you mind filing a bug at https://crbug.com/ so I remember to follow up on Monday?

I think, though, that we should consider splitting the bad ports list into those that we can block completely, and those that we can block iff the top-level browsing context doesn't match. 9100, for instance, looks like it's going to be a bit too widely used for us to block entirely, but might be blockable as a subresource request from the rest of the web (which would at least make the scanning mechanism described above a little harder to implement).

**Comment by annevk** [comment]
That sounds like an additional blocklist? No need to split the existing set.

**Comment by mikewest** [comment]
Yes, sorry. I mean that we'd conceptually split things out into "ports we can block completely", and "ports we can block for subresource requests". The latter would indeed be a new list that we'd (hopefully) be able to fill without as much risk to existing applications.

**Comment by zackw** [comment]
@mikewest Filed https://bugs.chromium.org/p/chromium/issues/detail?id=722108 (and, to be thorough, https://bugzilla.mozilla.org/show_bug.cgi?id=1364744) 

I like the idea of separate blacklists for top-level browsing contexts and subresource requests.

**Comment by annevk** [comment]
From https://en.wikipedia.org/wiki/Server_Message_Block it seems we should block 137, 138, and 445 for this.
